### PR TITLE
Profile Names

### PR DESCRIPTION
**Pull request by DreamyLynn**
Hey @SapuSeven :),

In the `OnCreate` Method of the `UserDatabase` class you create a Database with 
```kt
db.execSQL(UserDatabaseContract.Users.SQL_CREATE_ENTRIES_V3)
```
while the database version is already 4.
Is that intended?

### CODE REVIEW

**Comment by SapuSeven** [comment]
I _think_ that's actually a bug... But be sure to test if database upgrading still works correctly after changing it.

Edit: Oh, it seems you already handled that.

**Comment by SapuSeven** [comment]
I'm thinking about whether `profileName` should really be nullable... I don't think it would make sense to differentiate between `null` and being empty. You can always check using `isBlank()`. In that case it would simplify some things.

Any thoughts on this?

**Comment by DreamyLynn** [comment]
It is actually never null or blank anyways, except for when it is first initialized, since it is set to `user.userData.displayName` once a user is added and no `profileName` is supplied anyways. 
You're right though, initializing it as a blank string would remove the need for a non-null-asserted call whenever the value is required to not be null. What would also be possible, is to not set `profileName` to `displayName` and check if it is blank for every use, fall back to `displayName` in that case.
In hindsight my current solution could also cause the displayed name to not be changed, even though it is changed on Untis' side, so it's probably not ideal.
Checking all the values every time would be complicated too though, so having a function that returns the name to display would be useful.

**Comment by DreamyLynn** [comment]
So I think it's done.
Upgrade from develop branch and latest release works without any problems, all names are displayed correctly as far as I can tell.
And because I didn't mention that yet: Closes #120

File: app/src/main/java/com/sapuseven/untis/data/databases/UserDatabase.kt [hunk 1]
```diff
@@ -73,6 +73,13 @@ class UserDatabase private constructor(context: Context) : SQLiteOpenHelper(cont
 					db.execSQL("INSERT INTO ${UserDatabaseContract.Users.TABLE_NAME} SELECT * FROM ${UserDatabaseContract.Users.TABLE_NAME}_v3;")
 					db.execSQL("DROP TABLE ${UserDatabaseContract.Users.TABLE_NAME}_v3")
 				}
+				4 -> {
+					db.execSQL("ALTER TABLE ${UserDatabaseContract.Users.TABLE_NAME} RENAME TO ${UserDatabaseContract.Users.TABLE_NAME}_v4")
+					db.execSQL(UserDatabaseContract.Users.SQL_CREATE_ENTRIES_V5)
+					db.execSQL("ALTER TABLE ${UserDatabaseContract.Users.TABLE_NAME}_v4 ADD COLUMN ${UserDatabaseContract.Users.COLUMN_NAME_PROFILENAME}")
```
Comment (SapuSeven) [inline app/src/main/java/com/sapuseven/untis/data/databases/UserDatabase.kt]: Do you really need this `ALTER TABLE` statement?

```suggestion
```

**Review by SapuSeven — COMMENTED** [review]
Thanks, looks good to me now.
Just one minor thing; see below

File: app/src/main/java/com/sapuseven/untis/data/databases/UserDatabase.kt (hunk 1 above)
Comment (DreamyLynn) [reply app/src/main/java/com/sapuseven/untis/data/databases/UserDatabase.kt]: I don't, that was a remainder from a previous commit. Thank you!

**Review by SapuSeven — APPROVED** [review]

**Comment by SapuSeven** [comment]
Thanks a lot!

**Comment by SapuSeven** [comment]
Closes #120

**Pull request merged**